service/storage_proxy.cc

```diff
@@ -123,13 +123,17 @@ seastar::metrics::label_instance current_scheduling_group_label() {
 
 }
 
-[[maybe_unused]] static locator::host_id convert_ip_to_host_id_with_erm(locator::effective_replication_map_ptr ermptr,
+static locator::host_id convert_ip_to_host_id_with_erm(locator::effective_replication_map_ptr ermptr,
         const gms::inet_address& ep) {
     const auto& tm = ermptr->get_token_metadata();
     const auto& maybe_hid = tm.get_host_id_if_known(ep);
     if (maybe_hid) {
         return *maybe_hid;
     }
+    // During initialization of Scylla, there is a time when locator::token_metadata doesn't have any mappings yet
+    // or it is unstable (it removes and adds mappings IP<->host ID multiple times). However, this function provides
+    // a safe way to check if a given IP or host ID belongs to this node, even during that time. Hence this additional
+    // check.
```

Comment:
That use case was necessary in the previous version of the code because in `storage_proxy::create_write_response_handler()` we were explicitly creating a new vector of host IDs, obtaining them by mapping the IPs we got from the effective replication map. Then the vector was being passed to `cannot_hint()`. We have to perform queries to system tables during early stages of start-up, when topology is still unstable; because each mutation has to go through this function, if trying to map the IP of the local node failed (as it could!), Scylla would crash.

However, in the current version of this PR (where we do NOT create the vector anymore. We only create a view), and thanks to #16511 having been merged, I think the function is completely redundant. I don't know the semantics of Boost adaptors in detail, but I can modify the code so that it uses `std::ranges` to guarantee lazy evaluation of views. The function `cannot_hint()` first verifies that hinted handoff has been started (cf. the mentioned issue), so until that happens, we wouldn't need to perform any mapping at all, and there would be no problem.